var/spack/repos/builtin/packages/cosmomc/package.py

```diff
@@ -0,0 +1,197 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import fnmatch
+import os
+import shutil
+
+
+class Cosmomc(Package):
+    """CosmoMC is a Fortran 2008 Markov-Chain Monte-Carlo (MCMC) engine
+       for exploring cosmological parameter space, together with
+       Fortran and python code for analysing Monte-Carlo samples and
+       importance sampling (plus a suite of scripts for building grids
+       of runs, plotting and presenting results)."""
+
+    homepage = "http://cosmologist.info/cosmomc/"
+    url      = "https://github.com/cmbant/CosmoMC/archive/Nov2016.tar.gz"
+
+    version('2016.11', '98620cb746352f68fb0c1196e9a070ac')
+    version('2016.06', '92dc651d1407cca6ea9228992165f5cb')
+
+    def url_for_version(self, version):
+        names = {'2016.11': "Nov2016",
+                 '2016.06': "June2016"}
```

Comment:
I was curious about this so I did a quick check:
```
$ spack python
Spack version 0.10.0
Python 2.6.6, Linux x86_64
>>> from spack.version import *
>>> Version('2016.06') == Version('2016.6')
True
>>> Version('2016.5') < Version('2016.06') < Version('2016.7')
True
```
Looks like Spack should sort correctly regardless of whether we use leading zeros or not. Good to know!